Unify gophercloud revision

Frassle opened the pull request:
Projects that use dep (https://github.com/golang/dep/) such as pulumi (https://github.com/pulumi) can't use terraform-provider-openstack as a dependency due to a limitation in dep.

dep only supports one revision per project (i.e. git repo). Currently terraform-provider-openstack uses a lots of different revisions from gophercloud depending on which package is being imported. 

vgo (https://github.com/golang/vgo), which looks to be the successor to dep and will be in go 1.11, also only supports one revision per module. By default the whole repo is the module, so projects using vgo would also not be able to depend on terraform-provider-openstack and it's use of multiple revisions of gophercloud.

This PR simply unifies them all to the same revision 19abc56a8cd8d3630ab9735173cb92c0bf635f33, and fixes up one type mismatch in openstack/resource_openstack_compute_instance_v2.go due to the update.

Frassle commented:
TestAccObjectStorageV1Object_basic failed.

    Error creating OpenStack container object: Expected HTTP response code [201 202] when accessing [PUT http://192.168.0.9:8080/v1/AUTH_8930259935474be48aeb71c68ee1a1fe/tf_test_container_1/terraform/test/myfile.txt], but got 422 instead

objectstorage packages we're updated (were originally on e7fa81ec4cbb506026fa90fe7deeb9b7548a0728 and 0f64da0e36de86a0ca1a8f2fc1b0570a0d3f7504). So could be a real issue, I'll see if I can work it out.

As for testing, just some very basic compute tests on ovh cloud.

jtopjian commented:
> I'll see if I can work it out.

Sounds good. Let me know if you get stuck and I'll take a look, too.

> As for testing, just some very basic compute tests on ovh cloud.

No problem.

I should be able to test all resources on various clouds (excluding VPN, but the VPN resources haven't had any changes in Gophercloud since the resources were implemented). I'm pretty confident this vendor merge is possible to do - I just want to make sure we don't blow anything up in the process :smile:

Frassle commented:
Looks like it might be due to https://github.com/gophercloud/gophercloud/issues/603. Same code area effected, same symptoms. Attempted to write a workaround but no idea if it's any good.

jtopjian commented:
@Frassle I tracked the issue down to https://github.com/gophercloud/gophercloud/commit/10244f9e8a6751d9c0ac1656edb8d56aaaf97c32#diff-6e0b5c5f4dbc64f8c4d9f0e4fd6740ea and created #361 to address this.

I vendored the commit specifically to 10244f9e8a6751d9c0ac1656edb8d56aaaf97c32 so it wouldn't pull in any other changes.

Frassle commented:
Thanks @jtopjian for finding that! I've removed the commit I added to try and workaround the wrong problem. Let me know if I need to rebase this PR or anything else.

jtopjian commented:
@Frassle just a heads up that we found a core bug in #358 that will require a vendor update to the latest Gophercloud commit. 

In addition, #361 should be merged soon-ish.

Frassle commented:
I'll update the commit to refer to gophercloud@aac68cb146cb045a2c7c76ea060c81c9318f3dcd and await #361 to rebase ontop of.

jtopjian commented:
@Frassle Thanks!

I've begun creating a test report that I'll use to make sure we run all tests.

Frassle commented:
Well something with images is now borked. Following tests failed, TestAccOpenStackImagesV2ImageDataSource_basic, TestAccOpenStackImagesV2ImageDataSource_testQueries, TestAccImagesImageV2_importBasic, and TestAccImagesImageV2_basic. All with a error message stating "After applying this step, the plan was not empty". Again, I'll give it a look to see if can work it out, but help wouldn't go amiss.

jtopjian commented:
This is being tracked in #386 and #381.

The upstream Glance project has been merging patches which are causing conflicts with `properties`.

jtopjian commented:
#381 was already merged but coincidentally Glance added _more_ properties soon after, so that's what #386 is for. I have a new solution for this and I'll be opening a new PR for it. Sorry for the confusion.

For the time being, you can just ignore those errors in the test results.

But once it's all resolved I believe you will have to rebase. I don't think OpenLab is re-applying the PR on top of master. We can test this, though.

Frassle commented:
Rebase isn't a problem, looks like it will be needed for conflict resolution already anyway. Will watch #386 and push an update once that's merged.

jtopjian commented:
I just closed #386 in favor of #390. I'm quite confident #390 is the correct fix and is the PR to watch. Again, sorry for the confusion.

Frassle commented:
No problem will watch that PR.

Frassle commented:
Saw #390 was merged so rebased and updated.

jtopjian commented:
@Frassle Yep! I was going to wait until the acceptance tests were finished to confirm all problems are fixed. The tests have finished now and are all good.

Once your test results come in and confirmed passing, I'll start looking at manually running the omitted  tests.

jtopjian commented:
@Frassle quick update: I plan on doing the test inventory/audit within the next couple of days. It's at the top of my list - I just need a few hours set aside :)

jtopjian commented:
I've finished testing everything and it all looks good.

Thank you so much for the work you've done as well as your patience with this.

Frassle commented:
And thanks for your help with getting this merged :)
